src/main/java/Duke.java

```diff
@@ -1,10 +1,86 @@
+    private static Task[] tasks = new Task[100];
+    private static final String dividingLine = "\n————————————————————————————————————————————————\n";
+    public static void printList(){
+        System.out.println(dividingLine + "Here are the tasks in your list:");
+        for(int i = 0; i < taskCount; i++){
+            System.out.println((i + 1) +". " + tasks[i].toString());
+        }
+        System.out.println("Now you have " + taskCount + " tasks in the list." + dividingLine);
+    }
+    public static void addTask(Task task){
+        tasks[taskCount] = task;
+    }
     public static void main(String[] args) {
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke" + System.lineSeparator() + "What can I do for you?" + dividingLine);
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        while(!line.equals("bye")){
+            if(line.equals("list")){
+                printList();
+            }
+            else if(line.contains("mark") && !line.contains("unmark")){
+                int taskNum = Integer.parseInt(line.substring(5)) - 1;
+                tasks[taskNum].markAsDone();
+                System.out.println(dividingLine + "Nice! I've marked this task as done:");
+                System.out.println("[" + tasks[taskNum].getStatusIcon() + "] " + tasks[taskNum].description
+                        + dividingLine);
+            }
+            else if(line.contains("unmark")){
+                int unmarkNum = Integer.parseInt(line.substring(7)) - 1;
+                tasks[unmarkNum].unmarkAsDone();
+                System.out.println(dividingLine + "OK, I've marked this task as not done yet:");
+                System.out.println("[" + tasks[unmarkNum].getStatusIcon() + "] " + tasks[unmarkNum].description
+                        + dividingLine);
+            }
+            else if(line.contains("todo")){
+                line = line.replace("todo ", "");
+                addTask(new Todo(line));
+                System.out.println(dividingLine + "Got it. I've added this task:");
+                System.out.println(tasks[taskCount].toString());
+                System.out.println("Now you have " + (taskCount + 1) + " tasks in the list." + dividingLine);
+                taskCount++;
+            }
+            else if(line.contains("deadline")){
+                line = line.replace("deadline ", "");
+                addTask(new Deadline(line.substring(0, line.indexOf(" /") - 1),
+                        line.substring(line.indexOf("/by ") + 4)));
+                System.out.println(dividingLine + "Got it. I've added this task:");
+                System.out.println(tasks[taskCount].toString());
```

Comment:
If I am not mistaken, print(tasks[taskCount]) by default calls the toString() method of the Task object instantiation, hence no need to explicitly state the .toString() part.